### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by dragosv** [devcontainers/community#61] [Ideas]
Many projects require Windows as a development platform. Need to support Windows containers, not just Linux

**Comment by JohnLockwood** [comment]
Agreed -- it would appear this is [not supported](https://code.visualstudio.com/docs/devcontainers/containers#_getting-started) even for custom containers locally on VS Code, let alone on CodeSpaces.  It's a weird historical development -- Microsoft used to really want to get the developer experience right on Windows to grow its mindshare.  Now it develops or owns so many resources for open source work -- GitHub, VS Code, CodeSpaces, etc. -- but its own developers can go fish.  I suspect some pushback from the lawyers in the licensing department, but what do I know?  I just don't work here.

**Comment by RollsChris** [comment]
Yea i feel this is ridiculous, I cant use dev containers because  i work primarily with windows tech ;(. Ive also spent hours wasted trying to get it to work, so the fact that its not supported should be conveyed a lot better. Also is this not a lie: 
![image](https://github.com/devcontainers/community/assets/6059035/63674a9e-9a2f-422d-adeb-062720e4c3eb)

**Reply by ghost** [reply]
They are correct in that Windows can _host_ devcontainer. On Windows, there are two kinds of docker executors: one that runs Linux containers, another for windows containers. Apparently, for now devcontainer only supports the Linux container variant. The actual containers OS is listed below.

**Comment by ghost** [comment]
At work, we also have a tech-stack that would benefit from devcontainer for Windows containers.
As of right now, there is no matching base image for these, yet I am wondering what the reasons could be.

The base images themselves don’t seem to be all too special (just an ‘expected’ Dockerfile).

**Comment by matthewruzzi** [comment]
macOS/Xcode support also would be useful. 
https://github.com/orgs/devcontainers/discussions/211

**Comment by matthewruzzi** [comment]
**Related:**

Support for Windows Containers
https://github.com/orgs/community/discussions/9977

Request: Windows and macOS images
https://github.com/orgs/community/discussions/151996
